src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +135,36 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+
+  public void clean() {
+    try (final Connection connection = source.connect()) {
```

Comment:
hmm, interesting. clean() and writeRestOfQueue() are mutator methods (they cause something in the application to change). if you take a look at all the other functions in View, they are accessor functions, they only return part of the current state of the application without changing it. The correct place for these functions is with the other mutator functions in Controller.